v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,201 @@
+        (seedListType && seedListType !== "IDFA" && seedListType !== "GPADVID")
+      ) {
+        throw new CustomError(
+          `[Yahoo_DSP]:: seedListType is required for deviceId type audience and it should be any one of 'IDFA' and 'GPADVID'`,
+          400
+        );
+      }
+      outputPayload = createPayload(listDataKey, destination.Config);
+      // updatig seedListType here
+      outputPayload = {
+        ...outputPayload,
+        seedListType
+      };
+      break;
+    case "ipAddress":
+      outputPayload = createPayload(listDataKey, destination.Config);
+      // updating seedListType in outputPayload
+      outputPayload = {
+        ...outputPayload,
+        seedListType: "SHA256IP"
+      };
+      break;
+    case "mailDomain":
+      // traversing through every element in the add array for the elements to be added.
+      listDataKey.forEach(element => {
+        // storing keys of an object inside the add array.
+        const keys = Object.keys(element);
+        // For mailDomain the mailDomain or categoryIds must be present. throwing error if not present.
+        /**
+         * If mailDomain is not provided categoryIds is required. The audience includes consumers who have received
+         * mail from a domain belonging to the specified categories.
+         * Reference for use case of categoryIds:
+         * https://developer.yahooinc.com/dsp/api/docs/traffic/audience/mrt-audience.html#:~:text=Optional-,categoryIds,-Specifies%20an%20array
+         */
+        if (!(keys.includes(audienceType) || keys.includes("categoryIds"))) {
+          throw new CustomError(
+            `[Yahoo_DSP]:: Required property for ${audienceType} type audience is not available in an object`,
+            400
+          );
+        }
+        domains.push(element.mailDomain);
+        categoryIds.push(element.categoryIds);
+      });
+      outputPayload = { accountId, domains, categoryIds };
+      break;
+    case "pointOfInterest":
+      /**
+       * Here, populating includes and excludes object which contains POI locations visited by the consumer The consumer is
+       * added in the includes if the consumer has visited the POI location and is added in excludes if the
+       * consumer has not visited the POI.
+       */
+      outputPayload.includes = populateIncludes(listDataKey, audienceType);
+      outputPayload.excludes = populateExcludes(listDataKey, audienceType);
+      outputPayload = { ...outputPayload, accountId };
+      break;
+    default:
+      throw new CustomError(
+        `[Yahoo_DSP]:: Audience Type "${audienceType}" is not supported`,
+        400
+      );
+  }
+
+  const response = defaultRequestConfig();
+  response.endpoint = `${BASE_ENDPOINT}/traffic/audiences/${ENDPOINTS[audienceType]}/${audienceId}`;
+  response.body.JSON = removeUndefinedAndNullValues(outputPayload);
+  response.method = defaultPutRequestConfig.requestMethod;
+  const accessToken = await getAccessToken(destination);
```

Comment:
Already handling it in util from where access token returned.

Reply:
That is correct. But we are using optional chaining there so it might be possible for the function to return undefined. So we should handle it from here as well.

Reply:
Added